src/rocrate_inveniordm/mapping/processing_functions.py

```diff
@@ -96,3 +96,69 @@ def convert_to_iso_639_3(value):
         return None
 
     return code
+
+
+def typeProcessing(value):
+    """
+    Checks if the 'mainEntity' type includes 'ComputationalWorkflow'.
+    Works whether 'value' is a string or a list of strings.
+    """
+    if isinstance(value, str):
+        if "ComputationalWorkflow" in value:
+            return "workflow"
+    elif isinstance(value, list):
+        if any("ComputationalWorkflow" in v for v in value if isinstance(v, str)):
+            return "workflow"
+    return "dataset"
+
+
+def nameProcessing(value):
+    """
+    family_name is mandatory in Zenodo. Does not overwrite familyName and givenName
+    from the RO-Crate if they are provided, since this mapping is done before
+    """
+
+    if not value:
+        # With this we ensure that even if the author name is not mapped, at least
+        # its path index will be correctly set. That's why nameProcessing must be
+        # the first rule to be applied to support authors specified as strings
+        return {"type": "organizational", "name": ":unkn"}
+
+    new_name = {}
+    parts = value.strip().split()
+    new_name["family_name"] = parts[-1] if len(parts) > 0 else ""
+    new_name["given_name"] = " ".join(parts[:-1]) if len(parts) > 1 else ""
+    new_name["name"] = value
+    return new_name
+
+
+def rightsProcessing(value):
```

Comment:
I like the idea here, though what happens if someone puts in a non-SPDX URI? e.g. https://creativecommons.org/licenses/by/4.0/ . I am not sure that would give the right title ("Licences By 4.0"?)

I suspect it might be better to limit this to only be applied to SPDX URIs, and otherwise just pass the URI through. Do you know if the InvenioRDM/Zenodo API recognises other license ids? (The legacy Zenodo API only supported some SPDX IDs, see https://github.com/ResearchObject/ro-crate-zenodo/pull/15 and https://github.com/ResearchObject/ro-crate-zenodo/issues/5 for my earlier work with that)